src/main/java/uk/gov/justice/hmpps/prison/web/handler/ControllerAdvice.kt

```diff
@@ -0,0 +1,389 @@
+package uk.gov.justice.hmpps.prison.web.handler
+
+import jakarta.persistence.EntityExistsException
+import jakarta.validation.ValidationException
+import org.hibernate.exception.GenericJDBCException
+import org.slf4j.LoggerFactory
+import org.springframework.beans.TypeMismatchException
+import org.springframework.dao.DuplicateKeyException
+import org.springframework.dao.EmptyResultDataAccessException
+import org.springframework.dao.InvalidDataAccessApiUsageException
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.http.converter.HttpMessageConversionException
+import org.springframework.orm.jpa.JpaSystemException
+import org.springframework.security.access.AccessDeniedException
+import org.springframework.validation.FieldError
+import org.springframework.web.bind.MethodArgumentNotValidException
+import org.springframework.web.bind.MissingServletRequestParameterException
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.RestControllerAdvice
+import org.springframework.web.client.HttpClientErrorException
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestClientResponseException
+import uk.gov.justice.hmpps.prison.api.model.ErrorResponse
+import uk.gov.justice.hmpps.prison.service.BadRequestException
+import uk.gov.justice.hmpps.prison.service.ConflictingRequestException
+import uk.gov.justice.hmpps.prison.service.EntityAlreadyExistsException
+import uk.gov.justice.hmpps.prison.service.EntityNotFoundException
+import uk.gov.justice.hmpps.prison.service.NoContentException
+import java.sql.SQLException
+import java.util.stream.Collectors
+
+@RestControllerAdvice(basePackages = ["uk.gov.justice.hmpps.prison.api.resource"])
```

Comment:
was previously also including `"uk.gov.justice.hmpps.nomis.api.resource.controller"` unnecessarily as we don't have anything under there now